src/gobstuf/stuf/message.py

```diff
@@ -0,0 +1,61 @@
+from io import StringIO
+
+import os
+import xml.etree.ElementTree as ET
+
+from xml.dom import minidom
+
+
+class StufMessage:
+    """Workable representation of a StUF message, based on ElementTree.
+
+    If namespaces is not defined this class builds namespaces based on the input message to keep the representation
+    consistent and readable, and to make searching through a known XML string easier.
+    """
+
+    def __init__(self, msg: str, namespaces=None):
+        self.namespaces = namespaces
+        self.tree = None
+        self.load(msg)
+
+    def load(self, msg: str):
+        if not self.namespaces:
+            self.set_namespaces(msg)
+        self.tree = ET.fromstring(msg)
+
+    def set_namespaces(self, msg):
+        self.namespaces = dict([node for _, node in ET.iterparse(StringIO(msg), events=['start-ns'])])
+
+        for prefix, url in self.namespaces.items():
+            ET.register_namespace(prefix, url)
+
+    def find_elm(self, elements_str: str, tree=None):
+        if tree is None:
```

Comment:
In fact this is a recursive function that finds an arbitrary number of elements. find_elm is somewhat confusing, find_elements is more descriptive. I suggest adding some comment with an example to explain the behaviour.
Or split in get_elm and get_elements methods. Because the set_elm_value and get_elm_value do accept a parameter that is passed to set_elm but is not expected to contain spaces.

Reply:
No that's actually not the case. It always returns the first element (there was a find_elms method, but as it wasn't used I've removed it again). It always returns 1 element (or None).

I'll update the method comment to avoid confusion.